internal/releaseversion/releaseversion.go

```diff
@@ -56,23 +68,57 @@ func Validate(ctx context.Context, ctrlCLient client.Client, oldVersion semver.V
 	return nil
 }
 
-func availableReleases(ctx context.Context, ctrlClient client.Client) ([]*semver.Version, error) {
+func availableReleases(ctx context.Context, ctrlClient client.Client) ([]*semver.Version, []v1alpha1.Release, error) {
 	releaseList := &v1alpha1.ReleaseList{}
 	err := ctrlClient.List(ctx, releaseList)
 	if err != nil {
-		return []*semver.Version{}, microerror.Mask(err)
+		return []*semver.Version{}, nil, microerror.Mask(err)
 	}
 
 	var ret []*semver.Version
+	var releases []v1alpha1.Release
 	for _, release := range releaseList.Items {
 		parsed, err := semver.ParseTolerant(release.Name)
 		if err != nil {
-			return []*semver.Version{}, microerror.Maskf(errors.InvalidReleaseError, "Unable to parse release %s to a semver.Release", release.Name)
+			return []*semver.Version{}, nil, microerror.Maskf(errors.InvalidReleaseError, "Unable to parse release %s to a semver.Release", release.Name)
 		}
 		ret = append(ret, &parsed)
+		releases = append(releases, release)
 	}
 
-	return ret, nil
+	return ret, releases, nil
+}
+
+func filterOutAlphaAndIgnoredReleases(releases []*semver.Version, releaseCRs []v1alpha1.Release) ([]*semver.Version, []v1alpha1.Release) {
```

Comment:
Here, similar to `isOldOrNewReleaseIgnored()` below:

It seems that `releases` and `releaseCRs` must be sorted in the same order. Could `releases` be dropped and `releaseCRs[i].Name` be used instead? In this function only string of release version is needed and below the `semver` representation could be generated on-the-fly from `Release` object.

WDYT?

Reply:
Makes sense :+1:

Reply:
> It seems that releases and releaseCRs must be sorted in the same order. 

They are, as they are created in the same function at the same time.

Reply:
Ok, I took a look now, and some parts of the validation are using `semver.Version` objects, needed for release version comparison (greater than, less than, so actual `Version` struct is needed).

I added `Release` CR there so that annotation can be checked for ignored releases.

I initially left usage of both slices on purpose so that we don't have to parse version name to `semver.Version` object multiple times.